app/scripts/metamask-controller.js

```diff
@@ -2000,6 +2000,11 @@ export default class MetamaskController extends EventEmitter {
         ],
       }),
       trace,
+      decodingApiUrl: process.env.DECODING_API_URL,
+      isDecodeSignatureRequestEnabled: () =>
+        this.preferencesController.state.useExternalServices === true &&
```

Comment:
Do we usually couple API usage to multiple settings, transaction simulation for example doesn't use this?

Reply:
here condition is `isDecodeSignatureRequestEnabled` and thus I do not see a downside or adding preferences, env settings here